### ISSUE TITLE
num_sample in the sample_dataset function is overwritten

### ISSUE THREAD

**Issue by solomonno** [huggingface/setfit#230]
Hello!

`num_sample` argument in `sample_dataset` function does not work normally.
If the data is less than `num_samples` on a particular label, then less than `num_samples` will be sampled from the next label.

I think it's because `num_sample` is overwritten in the following code:
https://github.com/huggingface/setfit/blob/eee595ede962b9a9bbe62d4d919f5629d2fc2868/src/setfit/data.py#L100-L103

how about changing the variable name to the following code:
```python
 for label in range(num_labels): 
     data = shuffled_dataset.filter(lambda example: int(example[label_column]) == label) 
     num_label_samples = min(len(data) // 2, num_samples) 
     samples.append(data.select([i for i in range(num_label_samples)])) 
```

**Comment by tomaarsen** [comment]
Hello @solomonno,

Thank you for raising this! This seems to indeed be a bug. I've implemented your solution into #231 and listed you as co-author.

- Tom Aarsen

**Comment by solomonno** [comment]
Hi @tomaarsen! I confirmed your PR!

Thank you for it!

* Sungjun Kwon
